test/end-to-end/helpers/JSWait.js

```diff
@@ -1,46 +1,50 @@
 class JSWait extends codecept_helper {
 
     _beforeStep(step) {
+
         const helper = this.helpers.WebDriver || this.helpers.Puppeteer;
 
-        // Wait for content to load before checking URL
         if (step.name === 'seeCurrentUrlEquals' || step.name === 'seeInCurrentUrl') {
-            return helper.wait(3);
+            return helper.waitForElement('body', 30);
         }
+        return Promise.resolve();
     }
 
-    async navByClick(text, locator = null, webDriverWait = 2) {
+    async navByClick (text, locator) {
         const helper = this.helpers.WebDriver || this.helpers.Puppeteer;
         const helperIsPuppeteer = this.helpers.Puppeteer;
 
         if (helperIsPuppeteer) {
-            await Promise.all([
-                helper.page.waitForNavigation({waitUntil: ['domcontentloaded', 'networkidle0']}),
-                locator ? helper.click(text, locator) : helper.click(text)
-            ]);
-            return;
+            helper.click(text, locator).catch(err => {
+                console.error(err.message);
+            });
+            await helper.page.waitForNavigation({waitUntil: 'networkidle0'});
+        } else {
+            await helper.click(text, locator).catch(err => {
+                console.error(err.message);
+            });
+            await helper.wait(2);
         }
-        // non Puppeteer
-        await helper.click(text, locator);
-        await helper.wait(webDriverWait);
     }
-
-    async amOnLoadedPage(url) {
+    async amOnLoadedPage (url, language ='en') {
+        let newUrl = `${url}?lng=${language}`;
         const helper = this.helpers.WebDriver || this.helpers.Puppeteer;
         const helperIsPuppeteer = this.helpers.Puppeteer;
 
         if (helperIsPuppeteer) {
-            if (url.indexOf('http') !== 0) {
-                url = helper.options.url + url;
+            if (newUrl.indexOf('http') !== 0) {
+                newUrl = helper.options.url + newUrl;
             }
 
-            await Promise.all([
-                helper.page.waitForNavigation({waitUntil: ['domcontentloaded', 'networkidle0']}), // The promise resolves after navigation has finished
-                helper.page.goto(url)
-            ]);
+            helper.page.goto(newUrl).catch(err => {
+                console.error(err.message);
+            });
+            await helper.page.waitForNavigation({waitUntil: 'networkidle0'});
+
         } else {
-            await helper.amOnPage(url);
-            await helper.waitInUrl(url);
+            await helper.amOnPage(newUrl);
+            await helper.wait(2);
+            await helper.waitForElement('body');
```

Comment:
can this please wait for the URL to load, not just a 2 second wait, as this can cause ALOT of issues with crossbrowser test stability if pages take longer than 2 seconds to load (which they can often do). 
So like this:
```
    await helper.amOnPage(newUrl);
    await helper.waitInUrl(url);
    await helper.waitForElement('body');
```